In [945]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import re
import collections
from collections import defaultdict

# Loading the data

In [946]:
df_tweets = pd.read_csv("C:\\Users\\USER\\Documents\\Geetha\\Data Science\\Capstone Project 1\\tweets.csv", parse_dates=True, index_col='time')  

In [947]:
df_tweets.head(20)


name         username  \
time                                                  
2015-01-06 21:07:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:27:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:29:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:37:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:45:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:51:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 22:04:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 22:06:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 22:17:00  GunsandCoffee  GunsandCoffee70   
2015-01-10 00:05:00  GunsandCoffee  GunsandCoffee70   
2015-01-10 00:08:00  GunsandCoffee  GunsandCoffee70   
2015-01-12 23:10:00  GunsandCoffee  GunsandCoffee70   
2015-01-22 20:14:00  GunsandCoffee  GunsandCoffee70   
2015-01-22 20:47:00  GunsandCoffee  GunsandCoffee70   
2015-02-09 10:54:00  GunsandCoffee  GunsandCoffee70   
2015-02-12 13:03:00  GunsandCoffee  GunsandCoffee70   
2015-02-12 13:11:00  GunsandCoffee  GunsandCoffee70   
2015-02-16 15:57:00  GunsandCoffee  GunsandCoffee70   
2015-02-20 16:28:00  GunsandCoffee  GunsandCoffee70   
2015-02-20 16:29:00  GunsandCoffee  GunsandCoffee70   

                                                      description location  \
time                                                                         
2015-01-06 21:07:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-06 21:27:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-06 21:29:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-06 21:37:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-06 21:45:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-06 21:51:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-06 22:04:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-06 22:06:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-06 22:17:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-10 00:05:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-10 00:08:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-12 23:10:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-22 20:14:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-01-22 20:47:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-02-09 10:54:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-02-12 13:03:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-02-12 13:11:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-02-16 15:57:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-02-20 16:28:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   
2015-02-20 16:29:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews      NaN   

                     followers  numberstatuses  \
time                                             
2015-01-06 21:07:00        640              49   
2015-01-06 21:27:00        640              49   
2015-01-06 21:29:00        640              49   
2015-01-06 21:37:00        640              49   
2015-01-06 21:45:00        640              49   
2015-01-06 21:51:00        640              49   
2015-01-06 22:04:00        640              49   
2015-01-06 22:06:00        640              49   
2015-01-06 22:17:00        640              49   
2015-01-10 00:05:00        640              49   
2015-01-10 00:08:00        640              49   
2015-01-12 23:10:00        640              49   
2015-01-22 20:14:00        640              49   
2015-01-22 20:47:00        640              49   
2015-02-09 10:54:00        640              49   
2015-02-12 13:03:00        640              49   
2015-02-12 13:11:00        640              49   
2015-02-16 15:57:00        640              49   
2015-02-20 16:28:00        640              49   
2015-02-20 16:29:00        640              49   

 

In [948]:
df_tweets.info()


<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 17410 entries, 2015-01-06 21:07:00 to 2016-05-13 16:10:00
Data columns (total 7 columns):
name              17410 non-null object
username          17410 non-null object
description       14728 non-null object
location          11432 non-null object
followers         17410 non-null int64
numberstatuses    17410 non-null int64
tweets            17410 non-null object
dtypes: int64(2), object(5)
memory usage: 1.1+ MB


#### Columns 'description' and 'location' have null values

In [949]:
df_tweets['description'] = df_tweets['description'].fillna(' ')

In [950]:
df_tweets['location'] = df_tweets['location'].fillna(' ')

#### NaNs replaced with empty string in the columns 'description' and 'location' 

In [951]:
df_tweets.head(20)

name         username  \
time                                                  
2015-01-06 21:07:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:27:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:29:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:37:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:45:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:51:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 22:04:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 22:06:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 22:17:00  GunsandCoffee  GunsandCoffee70   
2015-01-10 00:05:00  GunsandCoffee  GunsandCoffee70   
2015-01-10 00:08:00  GunsandCoffee  GunsandCoffee70   
2015-01-12 23:10:00  GunsandCoffee  GunsandCoffee70   
2015-01-22 20:14:00  GunsandCoffee  GunsandCoffee70   
2015-01-22 20:47:00  GunsandCoffee  GunsandCoffee70   
2015-02-09 10:54:00  GunsandCoffee  GunsandCoffee70   
2015-02-12 13:03:00  GunsandCoffee  GunsandCoffee70   
2015-02-12 13:11:00  GunsandCoffee  GunsandCoffee70   
2015-02-16 15:57:00  GunsandCoffee  GunsandCoffee70   
2015-02-20 16:28:00  GunsandCoffee  GunsandCoffee70   
2015-02-20 16:29:00  GunsandCoffee  GunsandCoffee70   

                                                      description location  \
time                                                                         
2015-01-06 21:07:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-06 21:27:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-06 21:29:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-06 21:37:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-06 21:45:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-06 21:51:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-06 22:04:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-06 22:06:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-06 22:17:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-10 00:05:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-10 00:08:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-12 23:10:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-22 20:14:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-01-22 20:47:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-02-09 10:54:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-02-12 13:03:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-02-12 13:11:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-02-16 15:57:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-02-20 16:28:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            
2015-02-20 16:29:00  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews            

                     followers  numberstatuses  \
time                                             
2015-01-06 21:07:00        640              49   
2015-01-06 21:27:00        640              49   
2015-01-06 21:29:00        640              49   
2015-01-06 21:37:00        640              49   
2015-01-06 21:45:00        640              49   
2015-01-06 21:51:00        640              49   
2015-01-06 22:04:00        640              49   
2015-01-06 22:06:00        640              49   
2015-01-06 22:17:00        640              49   
2015-01-10 00:05:00        640              49   
2015-01-10 00:08:00        640              49   
2015-01-12 23:10:00        640              49   
2015-01-22 20:14:00        640              49   
2015-01-22 20:47:00        640              49   
2015-02-09 10:54:00        640              49   
2015-02-12 13:03:00        640              49   
2015-02-12 13:11:00        640              49   
2015-02-16 15:57:00        640              49   
2015-02-20 16:28:00        640              49   
2015-02-20 16:29:00        640              49   

 

In [952]:
df_tweets.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 17410 entries, 2015-01-06 21:07:00 to 2016-05-13 16:10:00
Data columns (total 7 columns):
name              17410 non-null object
username          17410 non-null object
description       17410 non-null object
location          17410 non-null object
followers         17410 non-null int64
numberstatuses    17410 non-null int64
tweets            17410 non-null object
dtypes: int64(2), object(5)
memory usage: 1.1+ MB


In [953]:
df_tweets=df_tweets.drop_duplicates()

In [954]:
df_tweets.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 17410 entries, 2015-01-06 21:07:00 to 2016-05-13 16:10:00
Data columns (total 7 columns):
name              17410 non-null object
username          17410 non-null object
description       17410 non-null object
location          17410 non-null object
followers         17410 non-null int64
numberstatuses    17410 non-null int64
tweets            17410 non-null object
dtypes: int64(2), object(5)
memory usage: 1.1+ MB


In [955]:
df_tweets.columns


Index(['name', 'username', 'description', 'location', 'followers',
       'numberstatuses', 'tweets'],
      dtype='object')

In [956]:
df_tweets['description'] = df_tweets['description'].str.lower()
df_tweets['tweets'] = df_tweets['tweets'].str.lower()

In [957]:
df_tweets.head(20)


name         username  \
time                                                  
2015-01-06 21:07:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:27:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:29:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:37:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:45:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 21:51:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 22:04:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 22:06:00  GunsandCoffee  GunsandCoffee70   
2015-01-06 22:17:00  GunsandCoffee  GunsandCoffee70   
2015-01-10 00:05:00  GunsandCoffee  GunsandCoffee70   
2015-01-10 00:08:00  GunsandCoffee  GunsandCoffee70   
2015-01-12 23:10:00  GunsandCoffee  GunsandCoffee70   
2015-01-22 20:14:00  GunsandCoffee  GunsandCoffee70   
2015-01-22 20:47:00  GunsandCoffee  GunsandCoffee70   
2015-02-09 10:54:00  GunsandCoffee  GunsandCoffee70   
2015-02-12 13:03:00  GunsandCoffee  GunsandCoffee70   
2015-02-12 13:11:00  GunsandCoffee  GunsandCoffee70   
2015-02-16 15:57:00  GunsandCoffee  GunsandCoffee70   
2015-02-20 16:28:00  GunsandCoffee  GunsandCoffee70   
2015-02-20 16:29:00  GunsandCoffee  GunsandCoffee70   

                                                      description location  \
time                                                                         
2015-01-06 21:07:00  english translations: http://t.co/qldj0ftews            
2015-01-06 21:27:00  english translations: http://t.co/qldj0ftews            
2015-01-06 21:29:00  english translations: http://t.co/qldj0ftews            
2015-01-06 21:37:00  english translations: http://t.co/qldj0ftews            
2015-01-06 21:45:00  english translations: http://t.co/qldj0ftews            
2015-01-06 21:51:00  english translations: http://t.co/qldj0ftews            
2015-01-06 22:04:00  english translations: http://t.co/qldj0ftews            
2015-01-06 22:06:00  english translations: http://t.co/qldj0ftews            
2015-01-06 22:17:00  english translations: http://t.co/qldj0ftews            
2015-01-10 00:05:00  english translations: http://t.co/qldj0ftews            
2015-01-10 00:08:00  english translations: http://t.co/qldj0ftews            
2015-01-12 23:10:00  english translations: http://t.co/qldj0ftews            
2015-01-22 20:14:00  english translations: http://t.co/qldj0ftews            
2015-01-22 20:47:00  english translations: http://t.co/qldj0ftews            
2015-02-09 10:54:00  english translations: http://t.co/qldj0ftews            
2015-02-12 13:03:00  english translations: http://t.co/qldj0ftews            
2015-02-12 13:11:00  english translations: http://t.co/qldj0ftews            
2015-02-16 15:57:00  english translations: http://t.co/qldj0ftews            
2015-02-20 16:28:00  english translations: http://t.co/qldj0ftews            
2015-02-20 16:29:00  english translations: http://t.co/qldj0ftews            

                     followers  numberstatuses  \
time                                             
2015-01-06 21:07:00        640              49   
2015-01-06 21:27:00        640              49   
2015-01-06 21:29:00        640              49   
2015-01-06 21:37:00        640              49   
2015-01-06 21:45:00        640              49   
2015-01-06 21:51:00        640              49   
2015-01-06 22:04:00        640              49   
2015-01-06 22:06:00        640              49   
2015-01-06 22:17:00        640              49   
2015-01-10 00:05:00        640              49   
2015-01-10 00:08:00        640              49   
2015-01-12 23:10:00        640              49   
2015-01-22 20:14:00        640              49   
2015-01-22 20:47:00        640              49   
2015-02-09 10:54:00        640              49   
2015-02-12 13:03:00        640              49   
2015-02-12 13:11:00        640              49   
2015-02-16 15:57:00        640              49   
2015-02-20 16:28:00        640              49   
2015-02-20 16:29:00        640              49   

 

In [958]:
df_tweets.name.value_counts(dropna=False)


Rami                    1475
War BreakingNews        1191
Conflict Reporter       1095
Salahuddin Ayubi        1056
Ibni Haneefah            709
                        ... 
Khilafah Superstate        7
Abdul05                    5
Ibn Al-Shishani            5
Newerajihadi61             4
fahad slyman alsomal       4
Name: name, Length: 112, dtype: int64

#### We are analyzing the tweets of 112 different users who have generated 17410 tweets

In [959]:

df_tweets['tweets'] =  df_tweets['tweets'] .apply(lambda elem: re.sub(r"(!|@|#|\n|')", "", elem)) 

#### Removing some of the special characters from 'tweets'column using regular expression operations

In [960]:
df_tweets['tweets'][6]



'english transcript : oh murabit : http://t.co/hujlj9kgkg http://t.co/t9ixmtbvgk'

In [961]:
df_tweets['tweets']

time
2015-01-06 21:07:00    english translation: a message to the truthful...
2015-01-06 21:27:00    english translation: sheikh fatih al jawlani f...
2015-01-06 21:29:00    english translation: first audio meeting with ...
2015-01-06 21:37:00    english translation: sheikh nasir al wuhayshi ...
2015-01-06 21:45:00    english translation: aqap: response to sheikh ...
                                             ...                        
2016-05-13 15:46:00    breakingislamicstate amaqagency fridaythe13th ...
2016-05-13 15:50:00    breaking iraqi forces bulldozer destroyed by g...
2016-05-13 15:51:00    breaking amaqagency islamicstate eastofhaditha...
2016-05-13 16:07:00    photoreport wilayatdimasiqh 📷the conduct of ba...
2016-05-13 16:10:00    amaqagency destroyed and burning iraqi forces ...
Name: tweets, Length: 17410, dtype: object

In [962]:
df_tweets['tweets_split'] = df_tweets['tweets'].str.split(' ')
df_tweets['tweets_split']

time
2015-01-06 21:07:00    [english, translation:, a, message, to, the, t...
2015-01-06 21:27:00    [english, translation:, sheikh, fatih, al, jaw...
2015-01-06 21:29:00    [english, translation:, first, audio, meeting,...
2015-01-06 21:37:00    [english, translation:, sheikh, nasir, al, wuh...
2015-01-06 21:45:00    [english, translation:, aqap:, response, to, s...
                                             ...                        
2016-05-13 15:46:00    [breakingislamicstate, amaqagency, fridaythe13...
2016-05-13 15:50:00    [breaking, iraqi, forces, bulldozer, destroyed...
2016-05-13 15:51:00    [breaking, amaqagency, islamicstate, eastofhad...
2016-05-13 16:07:00    [photoreport, wilayatdimasiqh, 📷the, conduct, ...
2016-05-13 16:10:00    [amaqagency, destroyed, and, burning, iraqi, f...
Name: tweets_split, Length: 17410, dtype: object

#### The tweets in 'tweets' column are splitted and are represened as a list in nw column 'tweets_split'

In [963]:

tweets_split = df_tweets['tweets_split'].sum()

In [964]:
tweets_split

['english',
 'translation:',
 'a',
 'message',
 'to',
 'the',
 'truthful',
 'in',
 'syria',
 '-',
 'sheikh',
 'abu',
 'muhammed',
 'al',
 'maqdisi:',
 'http://t.co/73xfszsjvr',
 'http://t.co/x8bzcscxzq',
 'english',
 'translation:',
 'sheikh',
 'fatih',
 'al',
 'jawlani',
 'for',
 'the',
 'people',
 'of',
 'integrity,',
 'sacrifice',
 'is',
 '',
 'easy',
 'http://t.co/uqqzxggvtz',
 'http://t.co/a7nbjwyhbr',
 'english',
 'translation:',
 'first',
 'audio',
 'meeting',
 'with',
 'sheikh',
 'fatih',
 'al',
 'jawlani',
 '(ha):',
 'http://t.co/tgxt1gdgw7',
 'http://t.co/zue8eisze6',
 'english',
 'translation:',
 'sheikh',
 'nasir',
 'al',
 'wuhayshi',
 '(ha),',
 'leader',
 'of',
 'aqap:',
 'the',
 'promise',
 'of',
 'victory:',
 'http://t.co/3qg5dkliwr',
 'http://t.co/7bqk1wjazc',
 'english',
 'translation:',
 'aqap:',
 'response',
 'to',
 'sheikh',
 'baghdadis',
 'statement',
 'although',
 'the',
 'disbelievers',
 'dislike',
 'it.',
 'http://t.co/2eym9eymte',
 'the',
 'second',
 'clip',
 '

In [965]:
count_stopwords = collections.Counter(tweets_split)

In [966]:
count_stopwords

Counter({'english': 38,
         'translation:': 10,
         'a': 2928,
         'message': 41,
         'to': 4197,
         'the': 7380,
         'truthful': 5,
         'in': 6589,
         'syria': 1162,
         '-': 430,
         'sheikh': 141,
         'abu': 340,
         'muhammed': 3,
         'al': 450,
         'maqdisi:': 1,
         'http://t.co/73xfszsjvr': 1,
         'http://t.co/x8bzcscxzq': 1,
         'fatih': 2,
         'jawlani': 9,
         'for': 1514,
         'people': 371,
         'of': 5919,
         'integrity,': 1,
         'sacrifice': 7,
         'is': 3316,
         '': 1374,
         'easy': 13,
         'http://t.co/uqqzxggvtz': 1,
         'http://t.co/a7nbjwyhbr': 1,
         'first': 126,
         'audio': 17,
         'meeting': 16,
         'with': 1359,
         '(ha):': 2,
         'http://t.co/tgxt1gdgw7': 1,
         'http://t.co/zue8eisze6': 1,
         'nasir': 15,
         'wuhayshi': 1,
         '(ha),': 1,
         'leader': 102,
    

In [967]:
stopwords_morethan50 = dict()
for (key, value) in ctr_df.items():
   # Check if key is even then add pair to new dictionary
   if value >= 50:
       stopwords_morethan50[key] = value
 

print(stopwords_morethan50)

{'a': 2928, 'to': 4197, 'the': 7380, 'in': 6589, 'syria': 1162, '-': 430, 'sheikh': 141, 'abu': 340, 'al': 450, 'for': 1514, 'people': 371, 'of': 5919, 'is': 3316, '': 1374, 'first': 126, 'with': 1359, 'leader': 102, 'statement': 57, 'by': 1672, 'soldier': 97, 'video': 271, ':': 254, 'please': 137, 'our': 225, 'new': 316, 'account': 71, 'after': 651, 'one': 423, 'was': 761, 'taken': 64, 'have': 746, 'been': 196, 'attack': 458, 'not': 749, 'it': 791, 'will': 662, 'if': 450, 'so': 316, 'we': 375, 'then': 170, 'let': 59, 'us': 754, 'way': 75, 'few': 72, 'more': 324, 'days': 125, 'very': 159, 'news': 178, 'reports': 253, 'and': 3400, 'syrian': 392, 'that': 967, 'should': 112, 'still': 176, 'or': 356, 'aleppo': 399, 'released': 96, 'group': 216, 'islamic': 635, 'do': 243, 'no': 297, 'rt': 5849, 'part': 103, '2': 299, '|': 231, 'و': 51, 'breaking': 267, 'state': 580, 'control': 170, 'border': 98, 'homs': 125, 'countryside': 84, 'im': 127, 'my': 245, 'best': 71, 'dawlah': 77, 'akhi': 109, 'he

In [968]:
tweets_http = df_tweets['tweets'].str.contains('https' ).sum()

In [969]:
tweets_http

8731

### Make a dictioary: Tokenize

In [970]:

list_words= ['turkey','islamicstate','syria','aleppo','warreporter1','russia', 'assad', 'iraq']
dict_token1=defaultdict(list)
for value in list_words:
   for key in ctr_df_iterated:
      if value in key:           
           dict_token1[value].append(key)
        
    
    

The words 'turkey','islamicstate','syria','aleppo','warreporter1','russia', 'assad', 'iraq' are found to be repeated in different formats

In [971]:
dict_token1

defaultdict(list,
            {'turkey': ['turkey'],
             'islamicstate': ['islamicstate'],
             'syria': ['syria', 'syrian', 'syria:', 'syria.', 'syria|n'],
             'aleppo': ['aleppo', 'aleppo.'],
             'warreporter1': ['warreporter1:', 'warreporter1'],
             'russia': ['russia', 'russian', 'russia|n'],
             'assad': ['assad', 'assads'],
             'iraq': ['iraq', 'iraqi', 'iraq|i']})